libtiledbsc/src/soma_collection.cc

```diff
@@ -0,0 +1,59 @@
+#include <regex>
+
+#include "tiledbsc/soma_collection.h"
+
+namespace tiledbsc {
+using namespace tiledb;
+
+//===================================================================
+//= public static
+//===================================================================
+SOMACollection SOMACollection::open(std::string_view uri) {
+    return SOMACollection(uri);
+}
+
+//===================================================================
+//= public non-static
+//===================================================================
+SOMACollection::SOMACollection(std::string_view uri) {
```

Comment:
In the Python impl I made a (perhaps overly lasagna-code)

```
SOMACollection extends TileDBGroup
SOMA extends TileDBGroup
AssayMatrix extends TileDBArray
...
TileDBGroup extends TileDBObject
TileDBArray extends TileDBObject
```

and I put things like trailing-slash strippers or what have you in the `TileDBObject` level -- food for thought ...

Reply:
I though about putting it in "utils", but saved that for another PR too.